app/jobs/readings_to_database_job.rb

```diff
@@ -0,0 +1,57 @@
+class ReadingsToDatabaseJob < ApplicationJob
+  retry_on ActiveRecord::Deadlocked, attempts: 10
+
+  def perform(key)
+    scan_readings_in_batches(key) do |batch|
+      process_batch(batch)
+      remove_readings(key, batch)
+    end
+  end
+
+  private
+
+  def batch_size
+    ArchiveConfig.READING_BATCHSIZE || 1000
```

Comment:
Can you add `READING_BATCHSIZE` to config.yml so we don't need to set a default here?